Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -0,0 +1,337 @@
+    init(profile: Profile) {
+        self.profile = profile
+        super.init(nibName: nil, bundle: nil)
+
+        self.profile.history.setTopSitesCacheSize(Int32(ASPanelUX.topSitesCacheSize))
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(TopSitesPanel.notificationReceived(_:)), name: NotificationFirefoxAccountChanged, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(TopSitesPanel.notificationReceived(_:)), name: NotificationProfileDidFinishSyncing, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(TopSitesPanel.notificationReceived(_:)), name: NotificationPrivateDataClearedHistory, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(TopSitesPanel.notificationReceived(_:)), name: NotificationDynamicFontChanged, object: nil)
+    }
+
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationFirefoxAccountChanged, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationProfileDidFinishSyncing, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationPrivateDataClearedHistory, object: nil)
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: NotificationDynamicFontChanged, object: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        refreshTopSites(ASPanelUX.topSitesCacheSize)
+        
+        reloadRecentHistoryWithLimit(ASPanelUX.historySize)
+
+        view.addSubview(tableView)
+        tableView.snp_makeConstraints { (make) in
+            make.edges.equalTo(self.view)
+        }
+    }
+
+    override func traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
+        self.topSiteHandler.currentTraits = self.traitCollection
+    }
+
+}
+
+// MARK: -  Section management
+extension ActivityStreamPanel {
+    enum Section: Int {
+        case topSites
+        case history
+
+        static let count = 2
+
+        var title: String? {
+            switch self {
+            case .history: return "Recent Activity"
+            default: return nil
+            }
+        }
+
+        var headerHeight: CGFloat {
+            switch self {
+            case .history: return 40
+            default: return 0
+            }
+        }
+
+        var headerView: UIView? {
+            switch self {
+            case .history:
+                let view = ASHeaderView()
+                view.title = "Recent Activity"
+                return view
+            default:
+                return nil
+            }
+        }
+
+        var cellIdentifier: String {
+            switch self {
+            case .topSites: return "TopSite"
+            default: return "Cell"
```

Comment:
Nit: For theses switch statements, prefer exhaustive cases instead of relying on default - even if we're returning nil. This way if we add anything to the enum we can lean on the compiler to yell at us instead of having behaviour we might not want.